### PR TITLE
Expand readme diagrams and documentation

### LINKED ISSUE #25: Expand README diagrams and documentation
Now that this repo is public, we should expand the README documentation to make it more comprehensive and accessible. Specifically, we should:

- [x] Create a Mermaid diagram or similar of how exactly the recurrent flagging works (pulls from what table, creates what flags, writes them to what table, etc.)
- [x] Add a table of the possible flag types, including their name in the `sv_outlier_type` column, the constraints required to create them, a rough count or percentage of sales of this type, etc.
- [x] A description of how this process will interact with human reviewers in the future
- [x] A diagram of how the output of this process is uploaded to iasWorld
- [x] Add a table of the possible outlier types, along with a *rough* breakdown of their commonality

### PR DESCRIPTION
**Pull request by wagnerlmichael**
Added "flags at a glance", and "next steps" sections. Let me know what needs more/less detail @dfsnow 

Closes #25

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,9 +8,11 @@ Table of Contents
 ================
 
 - [Overview](#overview)
+- [Flags at a Glance](#flags-at-a-glance)
```
Comment (dfsnow) [inline README.md:11]: **Nitpick:** For the sake of consistency with the other headings:

```suggestion
- [Flags at a glance](#flags-at-a-glance)
```

File: README.md [hunk 2]
```diff
@@ -8,9 +8,11 @@ Table of Contents
...
 - [Structure of data](#structure-of-data)
 - [Important flagging details](#important-flagging-details)
 - [AWS Glue job documentation](#aws-glue-job-documentation)
+- [Next Steps](#next-steps)
```
Comment (dfsnow) [inline README.md:15]:
```suggestion
- [Next steps](#next-steps)
```

File: README.md [hunk 3]
```diff
@@ -22,7 +24,9 @@ The workflow for sale flagging is as follows:
 * Next, `glue/sales_val_flagging.py` flags all new, unflagged sales. This script is automated such that it runs on a schedule (e.g. monthly).
 * If an error occurs or we want to update the methodology on previously-flagged sales, `manual_flagging/manual_update.py` is used to select a subset of sales to re-flag. All sales have a version number that is incremented on update. When utilizing our sales views, we pull the flag data with the highest version value to keep it up-to-date.
 
+#### Local Flagging
```
Comment (dfsnow) [inline README.md:27]: **Nitpick:** Add spaces after markdown headings, also consistent capitalization:

```suggestion
#### Local flagging

```

File: README.md [hunk 4]
```diff
@@ -22,7 +24,9 @@ The workflow for sale flagging is as follows:
 * Next, `glue/sales_val_flagging.py` flags all new, unflagged sales. This script is automated such that it runs on a schedule (e.g. monthly).
 * If an error occurs or we want to update the methodology on previously-flagged sales, `manual_flagging/manual_update.py` is used to select a subset of sales to re-flag. All sales have a version number that is incremented on update. When utilizing our sales views, we pull the flag data with the highest version value to keep it up-to-date.
```
Comment (dfsnow) [inline README.md:25]: **issue:** I would reframe these as the 3 possible run modes for the sales val pipeline, based on the following conditions:

1. No sales are flagged --> initial run
2. Some sales are already flagged --> glue job
3. Sales must be re-flagged --> manual update

Then update the diagrams accordingly.

File: README.md [hunk 5]
```diff
@@ -50,6 +54,96 @@ graph TD
...
+
+- **PTAX outlier (high)**: PTAX flag & [1 low statistical outlier type]
+- **Home flip sale (low)**: Short-term owner < 1 year & [1 low statistical outlier type]
+- **Family sale (low)**: Last name match & [1 low statistical outlier type]
+- **Non-person sale (low)**: Legal / corporate entity + [1 low statistical outlier type]
+- **Anomaly**: - Anomaly algorithm (low) & [1 low statistical outlier type]
+- **Low price (raw & sqft )**: Large swing away from mean & low price outlier 
+- **Low price swing**: Low price & low price per sq. ft. 
+- **Low price (raw)**: Low price (or under $10k)
+- **Low price (per sqft)**: Low price per sq. ft.
```
Comment (dfsnow) [inline README.md:118]: **suggestion:** I recommend you convert all of this into a markdown table. Right now it's difficult to read.

File: README.md [hunk 6]
```diff
@@ -50,6 +54,96 @@ graph TD
...
+    B --> C
+    C --> D
+    D -- Results saved to S3<br>with unique run ID --> E
+
+
+```
+
+# Flags at a Glance
+
+Sales from 2014 - present have been processed using our sales validation program. We assign flags to the following classes of properties:
```
Comment (dfsnow) [inline README.md:81]: **issue:** Add note about not flagging certain residential sales (multi-PIN, multi-card, etc.).

File: README.md [hunk 7]
```diff
@@ -50,6 +54,96 @@ graph TD
...
+|Low price (sqft)       |0.021     |
+|Anomaly (low)          |0.02      |
+|High price (sqft)      |0.019     |
+|Home flip sale (high)  |0.019     |
+|High price (raw & sqft)|0.016     |
+|Home flip sale (low)   |0.014     |
+|Family sale (low)      |0.006     |
+|Family sale (high)     |0.001     |
+|Low price swing        |~0.0       |
+|High price swing       |~0.0       |
```
Comment (dfsnow) [inline README.md:142]: **issue:** If you're going to include a table like this in the README, I recommend you also include the code required to rebuild it, since this could change significantly if we add new types, alter bounds, etc.

It might be prudent to create a Quarto README that has README.md as an output. You can see an example of this in any of the R packages i.e. https://github.com/ccao-data/assessr/blob/master/README.Rmd.

Note that you can use Python in a quarto doc, no need to build the table using R.

File: README.md [hunk 8]
```diff
@@ -50,6 +54,96 @@ graph TD
...
+- **Non-person sale (low)**: Legal / corporate entity + [1 low statistical outlier type]
+- **Anomaly**: - Anomaly algorithm (low) & [1 low statistical outlier type]
+- **Low price (raw & sqft )**: Large swing away from mean & low price outlier 
+- **Low price swing**: Low price & low price per sq. ft. 
+- **Low price (raw)**: Low price (or under $10k)
+- **Low price (per sqft)**: Low price per sq. ft.
+
+## Distribution of Outlier Types
+
+Around **7.2%** of the total sales have some sort of `Outlier` classification.  Within that 7.2%, the makeup of the outlier distribution is approximately as follows:
```
Comment (dfsnow) [inline README.md:122]: **issue:** This number should be dynamically updated. See note below.

File: README.md [hunk 9]
```diff
@@ -146,3 +241,49 @@ The Glue job and its flagging script are written in Python, while the job detail
...
+In the future, it is our goal that human sales validation analysts will be able to review these flags and manually override them. In the longer-term future we could train a supervised model using the analysts' judgement.
+
+
+
+
+
+
+
+
+
```
Comment (dfsnow) [inline README.md:289]: **nitpick:** Lots of unnecessary whitespace.

**Review by dfsnow — CHANGES_REQUESTED** [review]
There are a few major changes to make before this is good to go. Mostly I think we should include the code to generate count/proportion tables within the repo.

This README also needs some copy editing to make it clearer and more concise. I will do that in a separate PR and tag @wagnerlmichael for review.

File: README.md (hunk 7 above)
Comment (wagnerlmichael) [reply README.md]: Added query in html comment.

File: README.md (hunk 8 above)
Comment (wagnerlmichael) [reply README.md:145]: Added query in html comment.

File: README.md [hunk 10]
```diff
@@ -8,49 +8,231 @@ Table of Contents
 ================
```
Comment (dfsnow) [inline README.md:8]: **suggestion:** Let's drop the note and WIP blockquote (L3-L5).

File: README.md [hunk 11]
```diff
@@ -8,49 +8,231 @@ Table of Contents
...
 # Overview
 
 This repository contains code to identify and flag sales that may be non-arms-length transactions. A non-arms-length sale occurs when the buyer and seller have a relationship that might influence the transaction price, leading to a sale that doesn't reflect the true market value of the property. These sales can distort our analyses and models, since they don't adhere to the principle of an open and competitive market.
 
-The workflow for sale flagging is as follows:
 
-* A manual initial run of `manual_flagging/initial_flagging.py` instantiates all tables and flags all specified sales as either outliers or non-outliers.
-* Next, `glue/sales_val_flagging.py` flags all new, unflagged sales. This script is automated such that it runs on a schedule (e.g. monthly).
-* If an error occurs or we want to update the methodology on previously-flagged sales, `manual_flagging/manual_update.py` is used to select a subset of sales to re-flag. All sales have a version number that is incremented on update. When utilizing our sales views, we pull the flag data with the highest version value to keep it up-to-date.
+## Sales Validation Pipeline Run Modes
```
Comment (dfsnow) [inline README.md:22]: **nitpick:** Use consistent title capitalization. This should be:

```suggestion
## Sales validation pipeline run modes
```

File: README.md [hunk 12]
```diff
@@ -8,49 +8,231 @@ Table of Contents
...
 
-On the left, we see the normal workflow of the process. Represented on the right is the use of `manual_update.py` to update/re-flag sales.
+The Sales Validation (Sales Val) Pipeline can be executed in three distinct run modes, depending on the state of the sales data and the specific requirements for flagging:
```
Comment (dfsnow) [inline README.md:24]: **nitpick:** Sales Val Pipeline is pretty clunky. Let's do what we do in the residential and condo model README i.e.:

```suggestion
The Sales Validation Pipeline (hereafter referred to as “the pipeline”) can be executed in three distinct run modes, depending on the state of the sales data and the specific requirements for flagging:
```

File: README.md [hunk 13]
```diff
@@ -8,49 +8,231 @@ Table of Contents
...
+
+1. **Initial Run:** This mode is triggered when no sales have been flagged. It's the first step in the pipeline to instantiate tables and flag sales.
+2. **Glue Job:** This mode applies when there are already flagged sales in the system. It's an automated scheduled job that flags new, unflagged sales.
+3. **Manual Update:** This mode is used when sales need to be re-flagged, either due to errors or methodology updates. This allows for the selective re-flagging of sales.
```
Comment (dfsnow) [inline README.md:28]: **praise:** Nice, this is much clearer.

File: README.md [hunk 14]
```diff
@@ -8,49 +8,231 @@ Table of Contents
...
+
+### Initial Run Mode
+
+This is the starting point for the Sales Val Pipeline when no sales are flagged.
 
 ```mermaid
```
Comment (dfsnow) [inline README.md:34]: **suggestion:** Let's just consolidate the diagrams into a single diagram (so the diagrams are side-by-side).

File: README.md [hunk 15]
```diff
@@ -8,49 +8,231 @@ Table of Contents
 ================
 
 - [Overview](#overview)
+- [Flags at a glance](#flags-at-a-glance)
 - [Structure of data](#structure-of-data)
 - [Important flagging details](#important-flagging-details)
 - [AWS Glue job documentation](#aws-glue-job-documentation)
+- [Next steps](#next-steps)
 
 # Overview
```
Comment (dfsnow) [inline README.md:17]: For consistency with your other headings:

```suggestion
## Overview
```

File: README.md [hunk 16]
```diff
@@ -8,49 +8,231 @@ Table of Contents
...
+    D -->|Update process| F
+    C -->|New flag| E
+    E -->|Update process| F
+    F -->|Persist results| G
 ```
 
+
+## Flags at a Glance
+
+Sales from 2014 - present have been processed using our sales validation program. We flag properties from residential and condo data with the following specifications:
```
Comment (dfsnow) [inline README.md:96]: **suggestion:** I would add a note here about what you mean by "exclusions".

File: README.md [hunk 17]
```diff
@@ -8,49 +8,231 @@ Table of Contents
 ================
 
 - [Overview](#overview)
+- [Flags at a glance](#flags-at-a-glance)
 - [Structure of data](#structure-of-data)
 - [Important flagging details](#important-flagging-details)
```
Comment (dfsnow) [inline README.md:13]:
```suggestion
- [Flagging details](#important-flagging-details)
```

File: README.md (hunk 11 above)
Comment (dfsnow) [inline README.md:22]: **suggestion:** I would move this entire section under the **Flagging details** section. It's important, but the breakdown of what we're flagging and what the flags are is much _more_ important.

File: README.md [hunk 18]
```diff
@@ -8,49 +8,231 @@ Table of Contents
...
+    sv_outlier_type <> 'Not outlier'
+GROUP BY 
+    sv_outlier_type, total_count
+ORDER BY 
+    proportion DESC;
+-->
+
+|Outlier Type     |Proportion|
+|-----------------------|----------|
+|PTAX-203 flag          |0.401     |
```
Comment (dfsnow) [inline README.md:214]: **nitpick:** These should be percentages.

File: README.md (hunk 16 above)
Comment (wagnerlmichael) [reply README.md]: I added some context here but I could use your help with it. Not sure exactly what to add. @dfsnow

File: README.md [hunk 19]
```diff
@@ -106,16 +215,81 @@ erDiagram
...
+
+The Sales Validation Pipeline (hereafter referred to as “the pipeline”) can be executed in three distinct run modes, depending on the state of the sales data and the specific requirements for flagging:
+
+1. **Initial Run:** This mode is triggered when no sales have been flagged. It's the first step in the pipeline to instantiate tables and flag sales.
+2. **Glue Job:** This mode applies when there are already flagged sales in the system. It's an automated scheduled job that flags new, unflagged sales.
+3. **Manual Update:** This mode is used when sales need to be re-flagged, either due to errors or methodology updates. This allows for the selective re-flagging of sales.
+
+```mermaid
+
+graph TD
```
Comment (dfsnow) [inline README.md:228]: TIL that mermaid can use subgraphs...

File: README.md [hunk 20]
```diff
@@ -106,16 +215,81 @@ erDiagram
...
+
+        A1 -->|Initial setup| B1
+        B1 -->|Flag sales| C1
+        C1 -->|Store flags| D1
+        D1 -->|Persist results| E1
+    end
+
+    style A1 fill:#f9f,stroke:#333,stroke-width:2px
+    style B2 fill:#bbf,stroke:#333,stroke-width:2px
+    style A3 fill:#fbf,stroke:#333,stroke-width:2px
```
Comment (dfsnow) [inline README.md:279]: **issue:** These styles aren't readable (text is much too light compared to the fill color). Also, why have different colors for each subgraph?

**Review by dfsnow — APPROVED** [review]
Good enough for now! See my minor issue re: the styles for diagrams, etc. Once this is merged I'll do some additional copy editing.

File: README.md (hunk 20 above)
Comment (wagnerlmichael) [reply README.md]: I standardized the colors for the subgraphs, but I'm not sure what you mean about the text being light. Seems like the text is all black upon rendering? 

Do you mean that the fill colors are too dark for the text?
